compile error with gcc9.1 on linux

Compile error when building catch2, from the repository.

cloned catch2 yesterday (May 18,2019) with:
  git clone https://github.com/catchorg/Catch2.git
into a folder catch2.
Then started a build in that folder with 
  cmake .
  make

This produced the following error:
[ 99%] Building CXX object projects/CMakeFiles/SelfTest.dir/SelfTest/SurrogateCpps/catch_xmlwriter.cpp.o
[100%] Linking CXX executable SelfTest
/usr/bin/ld: cannot open output file SelfTest: Is a directory
collect2: error: ld returned 1 exit status
projects/CMakeFiles/SelfTest.dir/build.make:1703: recipe for target 'projects/SelfTest' failed
make[2]: *** [projects/SelfTest] Error 1
CMakeFiles/Makefile2:986: recipe for target 'projects/CMakeFiles/SelfTest.dir/all' failed
make[1]: *** [projects/CMakeFiles/SelfTest.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
alan@alan-SH67H3:~/projects/functional/code-examples/3rd-party/catchorg/Catch2$  

This looks like an error that was previously encountered and has resurrected itself.

**Platform information:**
This is on x86_64 with ubuntu 18.04.2
Using gcc 9.1.0
and cmake 3.14.4
and GNU make 4.1

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,10 @@ endif()
 
 project(Catch2 LANGUAGES CXX VERSION 2.7.2)
 
+if (CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
+    message(FATAL_ERROR "Building in-source is not supported! Create a build dir and remove ${CMAKE_SOURCE_DIR}/CMakeCache.txt")
+endif()
+
 # Provide path for scripts
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/CMake")
 

--- a/conanfile.py
+++ b/conanfile.py
@@ -18,7 +18,7 @@ class CatchConan(ConanFile):
         cmake.definitions["BUILD_TESTING"] = "OFF"
         cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
         cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
-        cmake.configure()
+        cmake.configure(build_folder='build')
         cmake.install()
 
         self.copy(pattern="LICENSE.txt", dst="licenses")

